code/modules/mob/living/carbon/carbon.dm

```diff
@@ -542,6 +542,34 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 
 /mob/living/carbon/throw_impact(atom/hit_atom, throwingdatum)
 	. = ..()
+	if(has_status_effect(STATUS_EFFECT_CHARGING))
+		var/hit_something = FALSE
+		if(ismovable(hit_atom))
+			var/atom/movable/AM = hit_atom
+			var/atom/throw_target = get_edge_target_turf(AM, dir)
+			if(!AM.anchored || ismecha(AM))
+				AM.throw_at(throw_target, 3, 14, src)
+				hit_something = TRUE
+		if(isobj(hit_atom))
+			var/obj/O = hit_atom
+			O.take_damage(150, BRUTE)
+			hit_something = TRUE
+		if(isliving(hit_atom))
+			var/mob/living/L = hit_atom
+			L.adjustBruteLoss(60)
+			L.Weaken(3)
+			shake_camera(L, 4, 3)
+			hit_something = TRUE
+		if(isturf(hit_atom))
+			var/turf/T = hit_atom
+			if(iswallturf(T))
+				T.dismantle_wall(TRUE)
+				hit_something = TRUE
+		if(hit_something)
+			visible_message("<span class='danger'>[src] slams into [hit_atom]!</span>", "<span class='userdanger'>[hit_atom] slams into you!</span>")
```

Comment:
`"You slam into [hit_atom]"` works a lot better IMO, since you're charging at them:
```suggestion
			visible_message("<span class='danger'>[src] slams into [hit_atom]!</span>", "<span class='userdanger'>You slam into [hit_atom]!</span>")
```
**Before:**
![image](https://user-images.githubusercontent.com/57483089/132143625-06081c72-34d4-4e0c-a4ca-3875fb8dd7b2.png)
**After:**
![image](https://user-images.githubusercontent.com/57483089/132143742-d1f62eba-7dd7-4dd3-939d-16159ffc0cde.png)